src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,305 @@
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> create(@PathVariable Long courseId, @RequestBody @Valid TutorialGroup tutorialGroup) throws URISyntaxException {
+        log.debug("REST request to create TutorialGroup: {} in course: {}", tutorialGroup, courseId);
+        if (tutorialGroup.getId() != null) {
+            throw new BadRequestException("A new tutorial group cannot already have an ID");
+        }
+
+        var course = courseRepository.findByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        tutorialGroup.setCourse(course);
+
+        trimStringFields(tutorialGroup);
+        checkTitleIsUnique(tutorialGroup);
+        TutorialGroup persistedTutorialGroup = tutorialGroupRepository.save(tutorialGroup);
+
+        return ResponseEntity.created(new URI("/api/tutorial-groups/" + persistedTutorialGroup.getId())).body(persistedTutorialGroup);
+    }
+
+    /**
+     * DELETE courses/:courseId/tutorial-groups/:tutorialGroupId : delete a tutorial group.
+     *
+     * @param courseId        the id of the course to which the tutorial groups belong to
+     * @param tutorialGroupId the id of the tutorial group to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> delete(@PathVariable Long courseId, @PathVariable Long tutorialGroupId) {
+        log.info("REST request to delete a TutorialGroup : {}", tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        tutorialGroupRepository.deleteById(tutorialGroupFromDatabase.getId());
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, tutorialGroupFromDatabase.getTitle())).build();
+    }
+
+    /**
+     * PUT /courses/:courseId/tutorial-groups : Updates an existing tutorial group
+     *
+     * @param courseId      the id of the course to which the tutorial group belongs
+     * @param tutorialGroup group the tutorial group to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated tutorial group
+     */
+    @PutMapping("/courses/{courseId}/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> update(@PathVariable Long courseId, @RequestBody @Valid TutorialGroup tutorialGroup) {
```

Comment:
This update endpoint is currently according to Artemis standard. 
We normally use a path like `PUT /api/entity/{entityId}` and then take the entityId as a parameter. We then check if the entityId equals the id of the request body.

Reply:
Done n 3b4418b238b4f4c15319f4688fc1d24d6f2897a3